course_discovery/populate_course_run_reporting_type_and_announcement.py

```diff
@@ -0,0 +1,114 @@
+import argparse
+import datetime
+from dateutil.parser import parse
+import json
+import logging
+import os
+
+import logging
+
+from edx_rest_api_client.client import EdxRestApiClient
+
+
+logger = logging.getLogger()
+
+
+class CatalogApiService(object):
+    """The service to interface with edX catalog API"""
+
+    def __init__(self, oauth_host, oauth_key, oauth_secret, api_url_root):
+        try:
+            self.access_token, expires = EdxRestApiClient.get_oauth_access_token(
+                '{root}/access_token'.format(root=oauth_host),
+                oauth_key,
+                oauth_secret, token_type='jwt'
+            )
+        except Exception:
+            logger.exception('No access token acquired through client_credential flow.')
+            raise
+
+        logger.info('Token retrieved: %s', access_token)
+
+        self.api_client = EdxRestApiClient(api_url_root, jwt=self.access_token)
+
+    def _get_resource_from_api(self, api_endpoint, page_size=None, **kwargs):
+        page = 1
+        results = []
+
+        while page >= 1:
+            if page_size:
+                response = api_endpoint.get(limit=page_size, offset=((page - 1) * page_size), **kwargs)
+            else:
+                response = api_endpoint.get(page=page, **kwargs)
+            if response.get('next'):
+                page += 1
+            else:
+                page = -1
+            results.extend(response.get('results'))
+
+        return results
+
+def process_line(line):
+    key, reporting_type, announcement = line.strip().split(',')
+    data = {
+        'reporting_type': reporting_type
+    }
+    if announcement != '':
+        # Convert 2/1/2016 to 2016-02-01T00:00:00+00:00
+        announcement = parse(announcement).isoformat()
+        data['announcement'] = announcement
+    return key, data
+
+def update_course_run(key, data, catalog_api_service):
+    course_runs_endpoint = catalog_api_service.api_client.course_runs(key)
+    try:
+        response = course_runs_endpoint.patch(data=data)
+        logging.info("{} update succeeded".format(key))
+    except Exception as e:
+        logging.error("Catalog returned this error: {}".format(e.response.content))
+        logging.error("{} update failed".format(key))
+
+def main():
+    parser = argparse.ArgumentParser(description='Populates reporting type and announcment on course runs')
+    parser.add_argument('--filename', help='Name of file with data on course run tags and announcment dates.')
+    parser.add_argument(
+        '--oauth_host',
```

Comment:
oauth_access_token_url

We only use this value to get the access token. Just get the actual access token endpoint instead of piecing it together.